hw_decorator.py

```diff
@@ -0,0 +1,99 @@
+import functools
+
+
+# ЗАДАЧА-1
+# Написать свой декоратор который будет проверять остаток от деления числа 100 на результат работы функции ниже.
+# Если остаток от деления = 0, вывести сообщение "We are OK!», иначе «Bad news guys, we got {}» остаток от деления.
+
+def divide_by_hundred(my_func):
+    """The decorator function which checks input for a valid type (int or float) and then checks if the result
+        of my_func is divided by 100 without rest or not."""
+    @functools.wraps(my_func)
+    def is_hundred_div(arg_one, arg_two):
+        if not isinstance(arg_one, int) or not isinstance(arg_one, int): #or
+            if not isinstance(arg_one, float) or not isinstance(arg_one, float):
+                raise ValueError("Only integer and float types are supported.")
+        result =  my_func(arg_one, arg_two) % 100
+        if not result:
+            print('We are OK!')
+        else:
+            print('Bad news guys, we got {0}'.format(result))
+    return is_hundred_div
+
+@divide_by_hundred
+def random_number_generator(arg_one, arg_two):
+    return arg_one * arg_two
+
+random_number_generator(25,12)
+random_number_generator(7.1, 25.5)
+
+
+# ЗАДАЧА-2
+# Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+def validate_input_decorator(my_func):
+    """The decorator function which checks the type of input argument."""
+    @functools.wraps(my_func)
+    def int_input_validator(int_input):
+        if isinstance(int_input, str):
+            raise ValueError("string type is not supported")
+        my_func(int_input)
+    return int_input_validator
```

Comment:
It works if we assume that only strings and integers are passed to the function. I would add "if isinstance(int_input, int):" check to be sure that no other types of data are put as function args.